### PR TITLE
IGNITE-20499 Broadcast user-provided initial cluster configuration with cluster state and save it to storage along with defaults

### PR DESCRIPTION
**Pull request by Flaugh24**
https://issues.apache.org/jira/browse/IGNITE-20499

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationChangerTest.java
Comment (ibessonov) [inline modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationChangerTest.java:590]: I believe there's an `assertArrayEquals` somewhere in JUnit

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java
Comment (ibessonov) [inline modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:243]: Please update the javadoc as well

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java
Comment (ibessonov) [inline modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:244]: Why are we declaring the field as `*Impl` anyway, is that intentional?

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java
Comment (ibessonov) [inline modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:800]: What if the entire lambda becomes `initializeClusterConfiguration`? Less lines, easier to read...

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java
Comment (ibessonov) [inline modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:1053]: I would recommend using EMPTY_CFG_SOURCE (or whatever that name is), it would simplify assertions in configuration changer.

File: modules/cluster-management/src/integrationTest/java/org/apache/ignite/internal/cluster/management/ItClusterManagerTest.java
Comment (tkalkirill) [inline modules/cluster-management/src/integrationTest/java/org/apache/ignite/internal/cluster/management/ItClusterManagerTest.java:117]: Can we then check the configuration itself and not the string representation?

File: modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationChangerTest.java
Comment (tkalkirill) [inline modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationChangerTest.java:580]: Please don't use `var` it's not clear what type it is.

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java
Comment (tkalkirill) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java:799]: There is not enough example in what form the configuration will be returned in string representation.

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java
Comment (ibessonov) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java:361]: Ok, how does it send it? I thought it would be done via replication, right?

File: modules/cluster-management/src/integrationTest/java/org/apache/ignite/internal/cluster/management/ItClusterManagerTest.java
Comment (ibessonov) [reply modules/cluster-management/src/integrationTest/java/org/apache/ignite/internal/cluster/management/ItClusterManagerTest.java]: Ok, I see

File: modules/configuration/README.md
Comment (ibessonov) [inline modules/configuration/README.md:357]:
```suggestion
Configuration defaults are defined in the configuration schema. However, it is not possible to define them there in the following cases:
```

File: modules/configuration/README.md
Comment (ibessonov) [inline modules/configuration/README.md:344]: Why did you change all paddings in the file?

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java
Comment (ibessonov) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java:325]: Please assert that the instance is not yet started. Just in case

**Review by ibessonov — APPROVED** [review]
Looks good to me. Let's wait for the approval of @tkalkirill

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**
